bcl2fastq/index_summary.py

```diff
@@ -89,14 +111,17 @@ def update_pruned_unknown_hidden(run_alias):
 )
 def update_known_index_bar(run_json):
     run = pandas.read_json(run_json, orient='split')
-
-    run['library'] = run['SampleID'].str.extract(
-        'SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_'
+        # convert a Json string into a panda object where orient is the expected format of json 
+    run['library'] = run['SampleID'].str.extract(   # create DataFrame via str.extract 
+        'SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_' 
     )
+        #replace index library with index sampleID as a string
```

Comment:
I'm not sure that this is quite accurate, but perhaps Savo can clarify

Reply:
This takes the `SampleID` column, extracts the library string out of each row value, and assigns those library strings to the new `library` column. 

As an example, from `SWID_11040179_NOVA_0003_Pb_R_PE_346_EX_NoGroup_171017_A00169_0049_AH53FCDMXX` the library string `NOVA_0003_Pb_R_PE_346_EX` gets extracted.